ssl/t1_lib.c

```diff
@@ -417,6 +417,320 @@ int ssl_load_groups(SSL_CTX *ctx)
+        goto err;
+    }
+
+    p = OSSL_PARAM_locate_const(params, OSSL_CAPABILITY_TLS_SIGALG_MAX_DTLS);
+    if (p == NULL || !OSSL_PARAM_get_int(p, &sinf->maxdtls)) {
+        ERR_raise(ERR_LIB_SSL, ERR_R_PASSED_INVALID_ARGUMENT);
+        goto err;
+    }
+    /*
+     * Now check that the algorithm is actually usable for our property query
+     * string. Regardless of the result we still return success because we have
+     * successfully processed this signature, even though we may decide not to
+     * use it.
+     */
+    ret = 1;
+    ERR_set_mark();
+    keymgmt = EVP_KEYMGMT_fetch(ctx->libctx, sinf->keytype?sinf->keytype:
+		    sinf->sig_name?sinf->sig_name:sinf->sigalg_name, ctx->propq);
+    if (keymgmt != NULL) {
+        /*
+         * We have successfully fetched the algorithm - however if the provider
+         * doesn't match this one then we ignore it.
+         *
+         * Note: We're cheating a little here. Technically if the same algorithm
+         * is available from more than one provider then it is undefined which
+         * implementation you will get back. Theoretically this could be
+         * different every time...we assume here that you'll always get the
+         * same one back if you repeat the exact same fetch. Is this a reasonable
+         * assumption to make (in which case perhaps we should document this
+         * behaviour)?
+         */
+        if (EVP_KEYMGMT_get0_provider(keymgmt) == provider) {
+            int keytype_nid;
+            /* We have a match - so we could use this signature */
+            /* Check proper object registration first, though */
+            /* don't care about return value as this may have been
+             * done within providers or previous calls to
+             * add_provider_sigalgs
+             */
+            OBJ_create(sinf->sigalg_oid, sinf->sigalg_name, NULL);
+	    if (sinf->sig_name)
+                OBJ_create(sinf->sig_oid, sinf->sig_name, NULL);
+	    if (sinf->keytype) {
+                keytype_nid = OBJ_create(sinf->keytype_oid, sinf->keytype, NULL);
+            }
+            else {
+                if (sinf->sig_name)
+                    keytype_nid = OBJ_txt2nid(sinf->sig_name);
+                else
+                    keytype_nid = OBJ_txt2nid(sinf->sigalg_name);
+            }
+	    if (sinf->hash_name) {
+                OBJ_create(sinf->hash_oid, sinf->hash_name, NULL);
+                if (sinf->sig_name)
+                    OBJ_add_sigid(OBJ_txt2nid(sinf->sig_name),
+                                  OBJ_txt2nid(sinf->hash_name),
+                                  keytype_nid);
+            }
+	    else {
+                OBJ_add_sigid(OBJ_txt2nid(sinf->sigalg_name),
+                              NID_undef,
+                              keytype_nid);
+            }
+	    /* sanity check: Without successful registration don't use alg */
+	    if ((OBJ_txt2nid(sinf->sigalg_name) == NID_undef) ||
```

Comment:
Nit: Replace tab with 8 spaces

Reply:
so changed in https://github.com/openssl/openssl/pull/19312/commits/1f20a63fbcb8f733651546e40328ae2ab2090537